vivisect/qt/main.py

```diff
@@ -201,6 +202,113 @@ def addBookmark(self, va, parent=None):
+        return self.views.get('VQVivMemoryView')
+
+    def getMemWidgetsByName(self, name='viv', firstonly=True):
+        '''
+        Returns a list of Memory View Widgets with the given name.
+        If "firstonly" is True, only return the first one or None(not a list)
+
+        Returns a tuple of (Widget, DockWidget).  The "Widget" is obtained from
+        the DockWidget, but they both have different powers.
+        '''
+        logger.debug("getWindowsByName(%r, firstonly=%r)", name, firstonly)
+        out = []
+
+        for vqDW in self.getMemoryWidgets():
+            w = vqDW.widget()
+            logger.debug("enn: " + w.getEnviNavName())
+            if w.getEnviNavName() == name:
+                if firstonly:
+                    return w, vqDW
+
+                out.append((w,vqDW))
+
+        if firstonly:   # if firstonly and we don't have one, return None
+            return None
+
+        return out
+
+    def getFuncGraphs(self):
+        return self.views.get('VQVivFuncgraphView')
+
+    def getFuncGraphsByName(self, name='FuncGraph0', firstonly=True):
+        '''
+        Returns a list of Dock Widgets which have a "getEnviNavName"
+        This includes MemoryViews and FuncGraphs
+        '''
+        logger.debug("getFuncGraphsByName()")
+
+        out = []
+        for vqDW in self.getFuncGraphs():
+            w = vqDW.widget()
+            logger.debug("enn: " + w.getEnviNavName())
+            if firstonly:
+                return w, vqDW
+
+            out.append((w, vqDW))
+
+        if firstonly:   # if firstonly and we don't have one, return None
+            return None
+
+        return out
+
+    def sendMemWidgetTo(self, va, wname='viv', firstonly=False):
+        '''
+        Tells the named Envi Nav Widget to navigate to the given VA
+        '''
+        logger.debug("sendMemWidgetsTo(0x%x, wname=%r)", va, wname)
+        for win in self.getMemWidgetsByName(wname, firstonly=False):
+            logger.debug("sendMemWidgetsTo gets %r", repr(win))
+            w, vqFW = win
+
+            logger.debug("sending %r to %r", w, hex(va))
+            w.enviNavGoto(hex(va))
+            if firstonly:
+                break
+        return True
+
+    def sendFuncGraphTo(self, va, wname='funcgraph0', firstonly=False):
+        '''
+        Tells the named Envi Nav Widget to navigate to the given VA
+        '''
+        logger.debug("sendFuncGraphTo(0x%x, wname=%r)", va, wname)
+        for win in self.getFuncGraphsByName(wname, firstonly=False):
+            logger.debug("sendFuncGraphTo gets %r", repr(win))
+            w, vqFW = win
+
+            logger.debug("sending %r to %r", w, hex(va))
+            w.enviNavGoto(hex(va))
+            if firstonly:
+                break
+        return True
+
+    def getCliBar(self):
+        '''
+        Returns the CLI Bar object
+        '''
+        logger.debug("getCliBar()")
+        for c in self.children():
+            if type(c) == vq_cli.VQCli:
```

Comment:
I'd prefer `isinstance` here just in case of subclasses, and anecdotally, it's faster.

Reply:
sure, makes sense.  done